# Amazon Sales Analysis

**Import libraries**

In [39]:
pip install plotly-express

In [40]:
pip install ydata-profiling

Note: you may need to restart the kernel to use updated packages.


In [79]:
import pandas as pd # used for data manipulation and data loading

import numpy as np # used numerical calculation

import matplotlib.pyplot as plt # used for visualization
%matplotlib inline

import seaborn as sns # advanced visulization

import plotly.express as px # designed for creating interactive visualizations with minimal code

from ydata_profiling import ProfileReport

**Load the dataset.**

In [81]:
df=pd.read_csv("C:\\Users\\Abc\\Desktop\\amazon.csv") # load the data set

**1.head()**: To get first rows in dataset.

In [45]:
df.head() # get first 5 rows

product_id                                       product_name  \
0  B07JW9H4J1  Wayona Nylon Braided USB to Lightning Fast Cha...   
1  B098NS6PVG  Ambrane Unbreakable 60W / 3A Fast Charging 1.5...   
2  B096MSW6CT  Sounce Fast Phone Charging Cable & Data Sync U...   
3  B08HDJ86NZ  boAt Deuce USB 300 2 in 1 Type-C & Micro USB S...   
4  B08CF3B7N1  Portronics Konnect L 1.2M Fast Charging 3A 8 P...   

                                            category discounted_price  \
0  Computers&Accessories|Accessories&Peripherals|...             ₹399   
1  Computers&Accessories|Accessories&Peripherals|...             ₹199   
2  Computers&Accessories|Accessories&Peripherals|...             ₹199   
3  Computers&Accessories|Accessories&Peripherals|...             ₹329   
4  Computers&Accessories|Accessories&Peripherals|...             ₹154   

  actual_price discount_percentage rating rating_count  \
0       ₹1,099                 64%    4.2       24,269   
1         ₹349                 43%      4       43,994   
2       ₹1,899                 90%    3.9        7,928   
3         ₹699                 53%    4.2       94,363   
4         ₹399                 61%    4.2       16,905   

                                       about_product  \
0  High Compatibility : Compatible With iPhone 12...   
1  Compatible with all Type C enabled devices, be...   
2  【 Fast Charger& Data Sync】-With built-in safet...   
3  The boAt Deuce USB 300 2 in 1 cable is compati...   
4  [CHARGE & SYNC FUNCTION]- This cable comes wit...   

                                             user_id  \
0  AG3D6O4STAQKAY2UVGEUV46KN35Q,AHMY5CWJMMK5BJRBB...   
1  AECPFYFQVRUWC3KGNLJIOREFP5LQ,AGYYVPDD7YG7FYNBX...   
2  AGU3BBQ2V2DDAMOAKGFAWDDQ6QHA,AESFLDV2PT363T2AQ...   
3  AEWAZDZZJLQUYVOVGBEUKSLXHQ5A,AG5HTSFRRE6NL3M5S...   
4  AE3Q6KSUK5P75D5HFYHCRAOLODSA,AFUGIFH5ZAFXRDSZH...   

                                           user_name  \
0  Manav,Adarsh gupta,Sundeep,S.Sayeed Ahmed,jasp...   
1  ArdKn,Nirbhay kumar,Sagar Viswanathan,Asp,Plac...   
2  Kunal,Himanshu,viswanath,sai niharka,saqib mal...   
3  Omkar dhale,JD,HEMALATHA,Ajwadh a.,amar singh ...   
4  rahuls6099,Swasat Borah,Ajay Wadke,Pranali,RVK...   

                                           review_id  \
0  R3HXWT0LRP0NMF,R2AJM3LFTLZHFO,R6AQJGUP6P86,R1K...   
1  RGIQEG07R9HS2,R1SMWZQ86XIN8U,R2J3Y1WL29GWDE,RY...   
2  R3J3EQQ9TZI5ZJ,R3E7WBGK7ID0KV,RWU79XKQ6I1QF,R2...   
3  R3EEUZKKK9J36I,R3HJVYCLYOY554,REDECAZ7AMPQC,R1...   
4  R1BP4L2HH9TFUP,R16PVJEXKV6QZS,R2UPDB81N66T4P,R...   

                                        review_title  \
0  Satisfied,Charging is really fast,Value for mo...   
1  A Good Braided Cable for Your Type C Device,Go...   
2  Good speed for earlier versions,Good Product,W...   
3  Good product,Good one,Nice,Really nice product...   
4  As good as original,Decent,Good one for second...   

                                      review_content  \
0  Looks durable Charging is fine tooNo complains...   
1  I ordered this cable to connect my phone to An...   
2  Not quite durable and sturdy,https://m.media-a...   
3  Good product,long wire,Charges good,Nice,I bou...   
4  Bought this instead of original apple, does th...   

                                            img_link  \
0  https://m.media-amazon.com/images/W/WEBP_40237...   
1  https://m.media-amazon.com/images/W/WEBP_40237...   
2  https://m.media-amazon.com/images/W/WEBP_40237...   
3  https://m.media-amazon.com/images/I/41V5FtEWPk...   
4  https://m.media-amazon.com/images/W/WEBP_40237...   

                                        product_link  
0  https://www.amazon.in/Wayona-Braided-WN3LG1-Sy...  
1  https://www.amazon.in/Ambrane-Unbreakable-Char...  
2  https://www.amazon.in/Sounce-iPhone-Charging-C...  
3  https://www.amazon.in/Deuce-300-Resistant-Tang...  
4  https://www.amazon.in/Portronics-Konnect-POR-1...

**2.tail()** : To get last rows in dataset.

In [83]:
df.tail() # to get last 5 rows

product_id                                       product_name  \
1460  B08L7J3T31  Noir Aqua - 5pcs PP Spun Filter + 1 Spanner | ...   
1461  B01M6453MB  Prestige Delight PRWO Electric Rice Cooker (1 ...   
1462  B009P2LIL4  Bajaj Majesty RX10 2000 Watts Heat Convector R...   
1463  B00J5DYCCA  Havells Ventil Air DSP 230mm Exhaust Fan (Pist...   
1464  B01486F4G6  Borosil Jumbo 1000-Watt Grill Sandwich Maker (...   

                                               category discounted_price  \
1460  Home&Kitchen|Kitchen&HomeAppliances|WaterPurif...             ₹379   
1461  Home&Kitchen|Kitchen&HomeAppliances|SmallKitch...           ₹2,280   
1462  Home&Kitchen|Heating,Cooling&AirQuality|RoomHe...           ₹2,219   
1463  Home&Kitchen|Heating,Cooling&AirQuality|Fans|E...           ₹1,399   
1464  Home&Kitchen|Kitchen&HomeAppliances|SmallKitch...           ₹2,863   

     actual_price discount_percentage rating rating_count  \
1460         ₹919                 59%      4        1,090   
1461       ₹3,045                 25%    4.1        4,118   
1462       ₹3,080                 28%    3.6          468   
1463       ₹1,890                 26%      4        8,031   
1464       ₹3,690                 22%    4.3        6,987   

                                          about_product  \
1460  SUPREME QUALITY 90 GRAM 3 LAYER THIK PP SPUN F...   
1461                       230 Volts, 400 watts, 1 Year   
1462  International design and styling|Two heat sett...   
1463  Fan sweep area: 230 MM ; Noise level: (40 - 45...   
1464  Brand-Borosil, Specification – 23V ~ 5Hz;1 W...   

                                                user_id  \
1460  AHITFY6AHALOFOHOZEOC6XBP4FEA,AFRABBODZJZQB6Z4U...   
1461  AFG5FM3NEMOL6BNFRV2NK5FNJCHQ,AGEINTRN6Z563RMLH...   
1462  AGVPWCMAHYQWJOQKMUJN4DW3KM5Q,AF4Q3E66MY4SR7YQZ...   
1463  AF2JQCLSCY3QJATWUNNHUSVUPNQQ,AFDMLUXC5LS5RXDJS...   
1464  AFGW5PT3R6ZAVQR4Y5MWVAKBZAYA,AG7QNJ2SCS5VS5VYY...   

                                              user_name  \
1460  Prabha ds,Raghuram bk,Real Deal,Amazon Custome...   
1461  Manu Bhai,Naveenpittu,Evatira Sangma,JAGANNADH...   
1462  Nehal Desai,Danish Parwez,Amazon Customer,Amaz...   
1463  Shubham Dubey,E.GURUBARAN,Mayank S.,eusuf khan...   
1464  Rajib,Ajay B,Vikas Kahol,PARDEEP,Anindya Prama...   

                                              review_id  \
1460  R3G3XFHPBFF0E8,R3C0BZCD32EIGW,R2EBVBCN9QPD9R,R...   
1461  R3DDL2UPKQ2CK9,R2SYYU1OATVIU5,R1VM993161IYRW,R...   
1462  R1TLRJVW4STY5I,R2O455KRN493R1,R3Q5MVGBRIAS2G,R...   
1463  R39Q2Y79MM9SWK,R3079BG1NIH6MB,R29A31ZELTZNJM,R...   
1464  R20RBRZ0WEUJT9,ROKIFK9R2ISSE,R30EEG2FNJSN5I,R2...   

                                           review_title  \
1460  Received the product without spanner,Excellent...   
1461  ok,everything was good couldn't return bcoz I ...   
1462  very good,Work but front melt after 2 month,Go...   
1463  Fan Speed is slow,Good quality,Good product,go...   
1464  Works perfect,Ok good product,Nice Product. Re...   

                                         review_content  \
1460  I received product without spanner,Excellent p...   
1461  ok,got everything as mentioned but the measuri...   
1462  plastic but cool body ,u have to find sturdy s...   
1463  I have installed this in my kitchen working fi...   
1464  It does it job perfectly..only issue is temp c...   

                                               img_link  \
1460  https://m.media-amazon.com/images/I/41fDdRtjfx...   
1461  https://m.media-amazon.com/images/I/41gzDxk4+k...   
1462  https://m.media-amazon.com/images/W/WEBP_40237...   
1463  https://m.media-amazon.com/images/W/WEBP_40237...   
1464  https://m.media-amazon.com/images/W/WEBP_40237...   

                                           product_link  
1460  https://www.amazon.in/Noir-Aqua-Spanner-Purifi...  
1461  https://www.amazon.in/Prestige-Delight-PRWO-1-...  
1462  https://www.amazon.in/Bajaj-RX-10-2000-Watt-Co...  
1463  https://www.amazon.in/Havells-Ve

**3.info()** : To get information about data.(i.e Null value,dtype)

In [85]:
df.info() # to get information about data

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1465 entries, 0 to 1464
Data columns (total 16 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   product_id           1465 non-null   object
 1   product_name         1465 non-null   object
 2   category             1465 non-null   object
 3   discounted_price     1465 non-null   object
 4   actual_price         1465 non-null   object
 5   discount_percentage  1465 non-null   object
 6   rating               1465 non-null   object
 7   rating_count         1463 non-null   object
 8   about_product        1465 non-null   object
 9   user_id              1465 non-null   object
 10  user_name            1465 non-null   object
 11  review_id            1465 non-null   object
 12  review_title         1465 non-null   object
 13  review_content       1465 non-null   object
 14  img_link             1465 non-null   object
 15  product_link         1465 non-null   object
dtypes: obj

**4.to_numeric** : This method is used to convert a Series or column of data to a numeric data type. This is particularly useful when working with data that may be stored as strings or objects but actually represents numeric values.

In [87]:
# Convert numerical columns  according to there datatype

# to convert the discounted price object to numerical column 
df['discounted_price'] = pd.to_numeric(df['discounted_price'],errors='coerce')  # errors=coerce is convert forcefully

#  to convert the actual price object to numerical column 
df['actual_price'] = pd.to_numeric(df['actual_price'],errors='coerce')

#  to convert the discount percentage object to numerical column 
df['discount_percentage'] = pd.to_numeric(df['discount_percentage'],errors='coerce')

# to convert the rating object to numerical column 
df['rating'] = pd.to_numeric(df['rating'],errors='coerce')

# # to convert the rating count object to numerical column 
df['rating_count'] = pd.to_numeric(df['rating_count'],errors='coerce')

**5.dtype** : To check data types of each column.

In [89]:
df.dtypes # to check data types

product_id              object
product_name            object
category                object
discounted_price       float64
actual_price           float64
discount_percentage    float64
rating                 float64
rating_count           float64
about_product           object
user_id                 object
user_name               object
review_id               object
review_title            object
review_content          object
img_link                object
product_link            object
dtype: object

**6.isnull()** : To check the missing values in each column.

In [91]:
df.isnull().sum() # to check missing values

product_id                0
product_name              0
category                  0
discounted_price       1465
actual_price           1465
discount_percentage    1465
rating                    1
rating_count           1139
about_product             0
user_id                   0
user_name                 0
review_id                 0
review_title              0
review_content            0
img_link                  0
product_link              0
dtype: int64

- In discounted_price , actual_price , discount_percentage are 1465 missing values are present.
- There is 1 missing value in rating and 1139 in rating_count.

**7.fillna()** : Fill missing values.

In [97]:
# fill missing values in discounted price by median
df['discounted_price'] = df['discounted_price'].fillna(df['discounted_price'].median())

# fill missing values in actual price by median 
df['actual_price'] = df['actual_price'].fillna(df['actual_price'].mean())

# fill missing values in rating by median
df['rating_count'] = df['rating_count'].fillna(df['rating_count'].mean())


#Fill missing  values in ratings with median 
df['rating'] = df['rating'].fillna(df['rating'].mean())


In [99]:
df.isnull().sum()

product_id                0
product_name              0
category                  0
discounted_price       1465
actual_price           1465
discount_percentage    1465
rating                    0
rating_count              0
about_product             0
user_id                   0
user_name                 0
review_id                 0
review_title              0
review_content            0
img_link                  0
product_link              0
dtype: int64